ext/lists.go

```diff
@@ -65,6 +90,26 @@ func (listsLib) CompileOptions() []cel.EnvOption {
 				}),
 			),
 		),
+		cel.Function("flatten",
+			cel.MemberOverload("list_flatten",
+				[]*cel.Type{listDyn}, listDyn,
+				cel.UnaryBinding(func(arg ref.Val) ref.Val {
+					list := arg.(traits.Lister)
+					flatList := flattenHelper(list, false)
+					return types.DefaultTypeAdapter.NativeToValue(flatList)
+				}),
+			),
+		),
+		cel.Function("flattenDeep",
+			cel.MemberOverload("list_flatten_deep",
+				[]*cel.Type{listDyn}, listDyn,
+				cel.UnaryBinding(func(arg ref.Val) ref.Val {
+					list := arg.(traits.Lister)
+					flatList := flattenHelper(list, true)
+					return types.DefaultTypeAdapter.NativeToValue(flatList)
+				}),
+			),
+		),
```

Comment:
Could you add check-time and runtime cost estimations? See https://github.com/google/cel-go/pull/850 for an example of how this was done and tests on `ext/sets.go` and `ext/sets_test.go`? This will help with onboarding into Kubernetes. Otherwise the `flatten()` will look like it has an infinite cost (which should only be true when someone uses `flatten(-1)`.